samples/logging/prometheus-grafana-opentelemetry/sample.md

```diff
@@ -0,0 +1,131 @@
+ * `nservicebus_failure_type` the exception type name (if applicable)
+
+## Exporting metrics
+
+The metrics are gathered using OpenTelemetry standard on the endpoint and need to be reported and collected by an external service. A Prometheus exporter can expose this data via an HTTP endpoint and the Prometheus service, hosted as a docker service, can retireve and store these information. This exporter is available via a NuGet package `OpenTelemetry.Exporter.Prometheus`. In this sample the service that exposes the data to scrape is hosted on `http://localhost:9185/metrics`. To enable the Prometheus exporter, the following should be executed:
+
+snippet: enable-prometheus-exporter
+
+Note: that the HTTP endpoint is also exposed through a local IP address so the Prometheus service running in docker can reach it over the network.
+
+The raw metrics retrieved through the scraping endpoint would look like this:
+
+```text
+# HELP nservicebus_messaging_successes Total number of messages processed successfully by the endpoint.
+# TYPE nservicebus_messaging_successes counter
+nservicebus_messaging_successes{nservicebus_discriminator="main",nservicebus_message_type="SomeCommand, Endpoint, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null",nservicebus_queue="OpenTelemetryDemo"} 850 1657693075515
+
+# HELP nservicebus_messaging_fetches Total number of messages fetched from the queue by the endpoint.
+# TYPE nservicebus_messaging_fetches counter
+nservicebus_messaging_fetches{nservicebus_discriminator="main",nservicebus_message_type="SomeCommand, Endpoint, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null",nservicebus_queue="OpenTelemetryDemo"} 1060 1657693075515
+
+# HELP nservicebus_messaging_failures Total number of messages processed unsuccessfully by the endpoint.
+# TYPE nservicebus_messaging_failures counter
+nservicebus_messaging_failures{nservicebus_discriminator="main",nservicebus_failure_type="System.Exception",nservicebus_message_type="SomeCommand, Endpoint, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null",nservicebus_queue="OpenTelemetryDemo"} 210 1657693075515
+```
+
+Below diagram shows the overall component interactions:
+
+```mermaid
+graph TD
+    A[NServiceBus Endpoint] -->|Report Metrics| B(Prometheus Exporter)
+    B -->|Expose| C{Metric Endpoint}
+    C -->|No Metrics| D[Status 200]
+    C -->|Has Metrics| E[Return Metrics]
+    F[Promethus Service] --> |Poll Metrics| E
+    F --> |Store Metrics| F
+    G[Grafana] --> |Query Data| F
+```
+
+## Docker stack
+
+Prometheus service needs to be configured to get the metrics data from the endpoint. Grafana also needs to be configured to get the data from Promethus and visualize it as graphs.
+
+To run the docker stack, run `docker-compse up -d` in the directory where the `docker-compose.yml` file is located.
+
+### Configuring Prometheus
+
+Copy the following files into the mapped volumes of the Prometheus and Grafana.
```

Comment:
can't we set that up in the sample so that this works without the user having to copy around files manually?

Reply:
We potentially can, but then they have manually go and define a new datasource in Grafana and define a scaper in Prometheus. This would go into too much detail in each specific solution. The older Prometheus + Grafana sample does the same as well.